include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -462,8 +462,8 @@ KOKKOS_INLINE_FUNCTION ddc::DiscreteElement<DDim> NonUniformBSplines<Tag, D>::
     std::array<double, degree()> left;
     std::array<double, degree()> right;
 
-    assert(x >= rmin());
-    assert(x <= rmax());
+    assert(x + Kokkos::abs(rmin()) * 1e-14 >= rmin());
```

Comment:
I would suggest to test a relative error like

```suggestion
    assert(Kokkos::abs(x - rmin()) < Kokkos::abs(rmin()) * 1E-14);
```

Reply:
I did the same with `length()` because if `rmin=0` no tolerancy is applied.

Reply:
Yes good idea. My test was wrong anyway it was checking near equality of `x` and `rmin()`